src/Microsoft.DotNet.ProjectJsonMigration/MSBuildExtensions.cs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Build.Evaluation;
+using Microsoft.Build.Construction;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Cli;
+using System.Linq;
+using System.IO;
+
+namespace Microsoft.DotNet.ProjectJsonMigration
+{
+    public static class MSBuildExtensions
+    {
+        public static IEnumerable<ProjectPropertyElement> PropertiesWithoutConditions(
+            this ProjectRootElement projectRoot)
+        {
+            return projectRoot.Properties
+                .Where(p => p.Condition == string.Empty
+                    && p.AllParents.All(parent => string.IsNullOrEmpty(parent.Condition)));
```

Comment:
Is there only one level that you care here or does AllParents finds all the ancestors in the tree?